src/test/java/com/thullo/service/BoardServiceImplTest.java

```diff
@@ -10,234 +10,67 @@
-
-
+    @InjectMocks
     private BoardServiceImpl boardServiceImplMock;
+    @Mock
+    private UserRepository userRepositoryMock;
+    @Mock
+    private UserPrincipal userPrincipalMock;
     private Board board;
-    private BoardRequest boardRequest;
-    private UserPrincipal userPrincipal;
-    String boardName = "DevDegree challenge";
-    String imageUrl = "http://localhost:8080/api/v1/thullo/files/123e4567-e89b-12d3-a456-426655440000";
 
     @BeforeEach
     void setUp() {
-        mapperMock = mock(ModelMapper.class);
-        boardRepositoryMock = mock(BoardRepository.class);
-        userRepositoryMock = mock(UserRepository.class);
-        fileServiceMock = mock(FileService.class);
-        roleServiceMock = mock(RoleServiceImpl.class);
-        notificationServiceMock = mock(NotificationServiceImpl.class);
-        boardServiceImplMock = new BoardServiceImpl(boardRepositoryMock, mapperMock, fileServiceMock, userRepositoryMock, roleServiceMock, notificationServiceMock);
-
+        userPrincipalMock = mock(UserPrincipal.class);
         board = new Board();
-        board.setName(boardName);
-
-        boardRequest = new BoardRequest();
-        boardRequest.setName(boardName);
-
-
-//        board.setTaskColumns(
-//                List.of(
-//                        new TaskColumn("Backlog \uD83E\uDD14", new Board()),
-//                        new TaskColumn("In Progress \uD83D\uDCDA", new Board()),
-//                        new TaskColumn("In Review ⚙️", new Board()),
-//                        new TaskColumn("Completed \uD83D\uDE4C\uD83C\uDFFD", new Board()))
-//
-//        );
-
-        userPrincipal = new UserPrincipal(
-                1L,
-                "Ismail Abdullah"
-                , "admin@gmail.com",
-                "password"
-                , true
-                , List.of(new SimpleGrantedAuthority("ROLE_USER"))
-        );
-
-        UpdateBoardRequest request = new UpdateBoardRequest();
-        request.setName("Thullo Challenge");
-        request.setBoardVisibility("public");
+        board.setName("Test Board");
+        board.setBoardVisibility(PUBLIC);
+        board.setUser(new User());
     }
 
     @Test
-    void testCreateBoard_withBoardName_createANewBoard() {
-        try {
-            when(userRepositoryMock.findByEmail(anyString()))
-                    .thenReturn(Optional.of(new User()));
-            when(mapperMock.map(boardRequest, Board.class))
-                    .thenReturn(board);
-            when(mapperMock.map(board, Board.class))
-                    .thenReturn(board);
-            when(boardRepositoryMock.save(board)).thenReturn(board);
+    public void testUpdateBoardWithValidRequest() throws UserException, BadRequestException, IOException {
 
-            BoardResponse actualResponse = boardServiceImplMock.createBoard(boardRequest, userPrincipal);
+        UpdateBoardRequest updateBoardRequest = new UpdateBoardRequest();
```

Comment:
you forget to set `updateBoardRequest.setBoardTag("")`, that was why the exception is being thrown